patroni/config.py

```diff
@@ -329,9 +329,21 @@ def reload_local_configuration(self) -> Optional[bool]:
 
     @staticmethod
     def _process_postgresql_parameters(parameters: Dict[str, Any], is_local: bool = False) -> Dict[str, Any]:
-        return {name: value for name, value in (parameters or {}).items()
-                if name not in ConfigHandler.CMDLINE_OPTIONS
-                or not is_local and ConfigHandler.CMDLINE_OPTIONS[name][1](value)}
+        pg_params: Dict[str, Any] = {}
+
+        for name, value in (parameters or {}).items():
+            if name not in ConfigHandler.CMDLINE_OPTIONS:
+                pg_params[name] = value
+            elif not is_local:
+                if ConfigHandler.CMDLINE_OPTIONS[name][1](value):
+                    pg_params[name] = value
+                else:
+                    default = ConfigHandler.CMDLINE_OPTIONS[name][0]
+                    if value != default:
```

Comment:
I guess `value == default` is impossible situation if validation failed.

Reply:
The `_false_validator` always returns `False` and would therefore always log.

I hit this situation when explicitly specifying `hot_standby=on` in our configuration, that *fails* validation currently.